rpcserver.go

```diff
@@ -1079,12 +1079,144 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
 	)
 	valid := err == nil
 
-	// TODO(roasbeef): also show additional final resting anchor
-	// information, etc?
+	decodedProof, err := r.marshalProofFile(ctx, proofFile, 0, false, false)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal proof: %w", err)
+	}
+
+	return &taprpc.VerifyProofResponse{
+		Valid:        valid,
+		DecodedProof: decodedProof,
+	}, nil
+}
+
+// DecodeProof attempts to decode a given proof file that claims to be anchored
+// at the specified genesis point.
+func (r *rpcServer) DecodeProof(ctx context.Context,
+	in *taprpc.DecodeProofRequest) (*taprpc.DecodeProofResponse, error) {
+
+	if len(in.RawProof) == 0 {
+		return nil, fmt.Errorf("proof file must be specified")
+	}
+
+	var proofFile proof.File
+	if err := proofFile.Decode(bytes.NewReader(in.RawProof)); err != nil {
+		return nil, fmt.Errorf("unable to decode proof file: %w", err)
+	}
+
+	latestProofIndex := uint32(proofFile.NumProofs() - 1)
+
+	if in.ProofAtDepth > latestProofIndex {
+		return nil, fmt.Errorf("invalid depth %d is greater than latest proof"+
```

Comment:
small nit: break the line after the space

```
		return nil, fmt.Errorf("invalid depth %d is greater than latest proof  "+
			" index of %d", in.ProofAtDepth, latestProofIndex)
```

Reply:
Goes over the 80 char limit that way :/

Reply:
It still does this way (remember to configure your IDE to use 8 spaces for the tab character). So just break before the word "latest"?